infrastructure/firecracker/config.go

```diff
@@ -65,7 +65,7 @@ func WithMicroVM(vm *models.MicroVM) ConfigOption {
 			ID:           vm.Spec.RootVolume.ID,
 			IsReadOnly:   vm.Spec.RootVolume.IsReadOnly,
 			IsRootDevice: true,
-			PathOnHost:   vm.Spec.RootVolume.MountPoint,
+			PathOnHost:   vm.Status.Volumes[vm.Spec.RootVolume.ID].Mount.Source,
```

Comment:
do we not want to do an `ok` check on that volume entry incase it is not mounted and added to the list?

Reply:
I notes that in the "Special notes for your reviewer" block in the pr description, but I can add it XD

Reply:
i noticed you added those later, sneaky 😉 

not too much to add 
```
status, ok := vm.Status.Volumes[vm.Spec.RootVolume.ID]
if !ok {
  return errors.NewVolumeNotMounted(vm.Spec.RootVolume.ID)
}
```
just before pleeeaaseee

Reply:
Added